xml/src/main/scala/akka/stream/alpakka/xml/scaladsl/XmlParsing.scala

```diff
@@ -40,7 +40,33 @@ object XmlParsing {
    */
   def parser(ignoreInvalidChars: Boolean = false,
              configureFactory: AsyncXMLInputFactory => Unit = configureDefault): Flow[ByteString, ParseEvent, NotUsed] =
-    Flow.fromGraph(new impl.StreamingXmlParser(ignoreInvalidChars, configureFactory))
+    Flow[ByteString].via(
+      Flow.fromGraph(
+        new impl.StreamingXmlParser[ByteString, ParseEvent, Unit](ignoreInvalidChars,
+                                                                  configureFactory,
+                                                                  getByteString = identity,
+                                                                  getContext = _ => (),
+                                                                  buildOutput = (parseEvent, _) => parseEvent)
```

Comment:
How about putting the three fixed transform methods into a class with two objects as implementations?

Reply:
Good idea. I did that in https://github.com/akka/alpakka/pull/2935/commits/2d980fb120140da7e9ba5eddcef5d45c9cc50e2d.